hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/io/storage/row/HoodieRowDataFileWriter.java

```diff
@@ -50,4 +56,12 @@ public interface HoodieRowDataFileWriter {
    * Closes the {@link HoodieRowDataFileWriter} and may not take in any more writes.
    */
   void close() throws IOException;
+
+  default void writeWithMetadata(HoodieKey key, HoodieRecord record, Schema schema, Properties props) throws IOException {
+    writeRow(key.getRecordKey(), (RowData) record.getData());
```

Comment:
does the row data already contain the metadata columns?

Reply:
yes